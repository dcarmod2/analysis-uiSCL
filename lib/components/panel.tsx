import {Box, Flex, useDisclosure} from '@chakra-ui/core'
import {faCaretDown, faCaretRight} from '@fortawesome/free-solid-svg-icons'

import Icon from './icon'

export function Panel({children}) {
  return (
    <Box borderWidth='1px' shadow='xs'>
      {children}
    </Box>
  )
}

export function Collapsible({defaultExpanded = true, children, heading}) {
  const {isOpen, onToggle} = useDisclosure(defaultExpanded)

  return (
    <Panel>
      <Heading onClick={onToggle} style={{cursor: 'pointer'}}>
        <Box>{heading}</Box>
        <Box>
          <Icon icon={isOpen ? faCaretDown : faCaretRight} />
        </Box>
      </Heading>
      {isOpen && children}
    </Panel>
  )
}

export function Heading({children, ...p}) {
  return (
    <Flex role='button' bg='gray.50' justify='space-between' p={4} {...p}>
      {children}
    </Flex>
  )
}

export function Body({children, ...p}) {
  return (
    <Box p={4} {...p}>
      {children}
    </Box>
  )
}
