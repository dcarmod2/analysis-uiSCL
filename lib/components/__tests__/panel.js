import enzyme from 'enzyme'
import React from 'react'

import {Collapsible, Body, Heading, Panel} from '../panel'

describe('Component > Panel', () => {
  describe('Panel', () => {
    it('renders correctly', () => {
      const tree = enzyme.mount(<Panel>Panel text</Panel>)
      expect(tree).toMatchSnapshot()
      tree.unmount()
    })
  })

  describe('Heading', () => {
    it('renders correctly', () => {
      const tree = enzyme.mount(<Heading>Heading text</Heading>)
      expect(tree).toMatchSnapshot()
      tree.unmount()
    })
  })

  describe('Body', () => {
    it('renders correctly', () => {
      const tree = enzyme.mount(<Body>Body text</Body>)
      expect(tree).toMatchSnapshot()
      tree.unmount()
    })
  })

  describe('Collapsible', () => {
    it('should expand / collapse correctly', () => {
      const tree = enzyme.mount(
        <Collapsible defaultExpanded={false} heading='Heading'>
          <div id='body' />
        </Collapsible>
      )
      expect(tree.exists('#body')).toBe(false)
      tree.find('Heading').simulate('click')
      expect(tree.exists('#body')).toBe(true)
    })
  })
})
