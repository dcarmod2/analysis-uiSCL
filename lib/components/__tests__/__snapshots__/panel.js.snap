// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Panel Body renders correctly 1`] = `
<Body>
  <Box
    p={4}
  >
    <div
      className="css-1wcaknn"
    >
      Body text
    </div>
  </Box>
</Body>
`;

exports[`Component > Panel Heading renders correctly 1`] = `
<Heading>
  <Flex
    bg="gray.50"
    justify="space-between"
    p={4}
    role="button"
  >
    <Box
      bg="gray.50"
      display="flex"
      justifyContent="space-between"
      p={4}
      role="button"
    >
      <div
        className="css-14uk9yj"
        role="button"
      >
        Heading text
      </div>
    </Box>
  </Flex>
</Heading>
`;

exports[`Component > Panel Panel renders correctly 1`] = `
<Panel>
  <Box
    borderWidth="1px"
    shadow="xs"
  >
    <div
      className="css-1loeamy"
    >
      Panel text
    </div>
  </Box>
</Panel>
`;
